code/lora/lora.py

```diff
@@ -0,0 +1,298 @@
+import time
+#TODO import argparse
+from pathlib import Path
+from SX127x.LoRa import *
+from SX127x.board_config import BOARD
+
+HOME_PATH = Path.home()
+LORA_LOGS = HOME_PATH / 'lora_logs'
+RX_LOG = LORA_LOGS / 'rx.txt' #File, where all received data is stored
+STATUS_LOG = LORA_LOGS / 'status.txt' #File, where this block's behavior is logged
+CONF_FILE = LORA_LOGS / 'conf.txt' #File, where commands for this block are issued by other scripts
+IMAGES = HOME_PATH / 'images'
+THERMAL = HOME_PATH / 'thermal'
+
+#Log files with data to be sent
+#TODO: Argparse or .conf of some sort
+TX_FILES = [
+    HOME_PATH / 'sensor_logs' / 'alpha.txt',
+    HOME_PATH / 'sensor_logs' / 'bravo.txt',
+    HOME_PATH / 'sensor_logs' / 'charlie.txt'
+]
+
+IMAGES_PATH = HOME_PATH / 'images'
+THERMAL_PATH = HOME_PATH / 'thermal'
+
+UP_PACKET_LENGTH = 11 #Base -> Sat packet length in bytes
+DOWN_PACKET_LENGTH = 25 #Sat -> Base packet length in bytes
+CMD_LENGTH = 10
+
+#DEFAULT transmission mode constants
+MAX_BACKREAD = 5 #Maximal amount of lines expected to be logged into a file within one TX window
+MAX_BUFFER_LENGTH = 50 #Maximal packet buffer length after which it overflows
+MAX_BUFFER_APPEND = MAX_BACKREAD*len(TX_FILES)
+TX_LINE_LENGTH = DOWN_PACKET_LENGTH-1
+
+TX_TIME = 0.11
+RX_TIME = 0.14
+
+THERMAL_HEIGHT = 32
+THERMAL_WIDTH = 24
+
+IMG_QUAL = {
+    'L': 'low',
+    'M': 'medium',
+    'H': 'high',
+    'U': 'ultra'
+}
+
+
+class HEADER: #Packet Headers
+    MSGPACK = "M"
+    IMAGE = "I"
+    THERMAL = "T"
+    STATUS = "S"
+    MSGPACK_B = 77
+    IMAGE_B = 73
+    THERMAL_B = 84
+    STATUS_B = 83
+    LORA = [255, 255, 0, 0]
+
+
+class myLoRa(LoRa):
+    def __init__(self, verbose=False):
+        super(myLoRa, self).__init__(verbose)
+        self.set_mode(MODE.SLEEP)
+        self.set_dio_mapping([0] * 6)
+        self.ack = {'thermal': 1, 'image':1} #Keeps track of whether an ACK was rececived
+        self.image_waiting = []
+        self.thermal_waiting = []
+        self.tx_end = False #Set to true to stop the script
+        self.filepos = {path: 0 for path in TX_FILES} #Tracks lengths of logfiles
+        self.conf_filepos = 0
+        self.buffer = [] #Temporarily stores log readings to be sent
+        self.sending_image = False
+        self.sending_thermal = False
+        self.thermal_counter = 0
+        self.image_size = 0
+        self.last_tx_mode = 'default'
+
+        with open(CONF_FILE, 'w') as cf:
+            cf.write("NULLCMD--\n") #write some crap to prevent negative seek()
+
+
+    def on_rx_done(self):
+        """
+        Triggers automatically upon LoRa interrupting received packets.
+        Reads the packet and logs it into a file.
+        """
+
+        self.clear_irq_flags(RxDone=1)
+        raw_rx_payload = self.read_payload(nocheck=True) #TODO fix that crc bug or implement high-layer crc
+        payload = bytes(raw_rx_payload).decode("utf-8", "ignore").strip('\x00')
+        with open(RX_LOG, "a") as rx_log:
```

Comment:
have you heard about [logging module](https://docs.python-guide.org/writing/logging/)? It does exactly that what we want I believe

Reply:
Maybe misnamed the files, nevertheless this one is a file used to communicate between LoRa and CommandBlock, hence the lib has no use here imho. But, if we want to store another human-readable copy of all data received by the can, this would pe perfect.

Reply:
Okay, I accept that we're find with working on raw files, thats cool. But please at least substitute all `print` statements with logging and we're good to merge 🏆